sql: support PostgreSQL's abbreviated time units

ivan opened the pull request:
I spotted one of these in the wild in [postgrex's test suite](https://github.com/elixir-ecto/postgrex/blob/51bfc230f2e8e85e539121102cb262fdc39a52b5/test/query_test.exs#L94), and I added the rest because they are probably in use elsewhere.

ivan committed:
sql: support PostgreSQL's abbreviated time units

Also support abbreviated time units for the nanosecond interval. Some
quick experimentation revealed that it was annoying not to have them
when they are available for the other units.

Release note (sql change): CockroachDB now recognizes PostgreSQL's
abbreviated time units when converting strings to intervals.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/27393)
<!-- Reviewable:end -->

knz commented:
Thank you! This change looks great.

Just two minor changes needed to match our PR requirements:

- because both commits pertain to the same unit of work, please squash them together. We adopt the reasoning 1 commit = 1 unit of work.

- because this is a user-facing change, the commit message should also include a release note. You can append the following text, for example:

```
Release note (sql change): CockroachDB now recognizes the same units of time as PostgreSQL when converting strings to intervals.
```

Thank you again

ivan commented:
Thanks for the review! Hopefully I addressed everything. I changed the wording of the release note slightly. They're not quite the same because CockroachDB has `{"µs", "μs"}`, nanoseconds, and is missing support for some time units.

knz commented:
Beautiful. Many thanks for your contribution!

bors r+
